spec/controllers/state_file/questions/return_status_controller_spec.rb

```diff
@@ -378,6 +378,52 @@
               expect(response.body).to include I18n.t("state_file.questions.return_status.rejected.next_steps.can_edit.body")
             end
           end
+
+          context "specified error-shown edge cases" do
+            let(:rejected_transition) { efile_submission.efile_submission_transitions.where(to_state: 'rejected').last }
+            let!(:transition_error) {
+              create :efile_submission_transition_error,
+                     efile_error: main_efile_error,
+                     efile_submission_transition: rejected_transition,
+                     efile_submission_id: efile_submission.id
+            }
+            let!(:related_transition_error) {
+              create :efile_submission_transition_error,
+                     efile_error: related_efile_error,
+                     efile_submission_transition: rejected_transition,
+                     efile_submission_id: efile_submission.id
+            }
+
+            context "with NCD400-1100 and NCD400-1080 errors" do
+              let(:main_efile_error) { create :efile_error, code: "NCD400-1100", service_type: :state_file_nc, expose: true, auto_wait: true }
+              let(:related_efile_error) { create :efile_error, code: "NCD400-1080", service_type: :state_file_nc, expose: true, auto_wait: true }
+
+              it "should show the NCD400-1100 even though NCD400-1080 is the last error" do
+                get :edit
+                expect(response.body).to include main_efile_error.code
+                expect(response.body).not_to include related_efile_error.code
+              end
+            end
+
+            context "with Form502-01550-010 and related errors" do
+              let(:main_efile_error) { create :efile_error, code: "Form502-01150-010", service_type: :state_file_md, expose: true, auto_wait: true }
+              let(:related_efile_error) { create :efile_error, code: "Form502-01230-005", service_type: :state_file_md, expose: true, auto_wait: true }
+              let(:related_efile_error_2) { create :efile_error, code: "Form502-01150-005", service_type: :state_file_md, expose: true, auto_wait: true }
+              let!(:related_transition_error_2) {
+                create :efile_submission_transition_error,
+                       efile_error: related_efile_error_2,
+                       efile_submission_transition: rejected_transition,
+                       efile_submission_id: efile_submission.id
+              }
```

Comment:
there are more than 2 related errors for MD, but thought 2 were enough -- lmk if we want to test all of them